### PR TITLE
fix: make MQTT connector error log messages easier to understand

### PR DESCRIPTION
**Pull request by kjellwinblad**
Fixes:
https://emqx.atlassian.net/browse/EMQX-12555
https://emqx.atlassian.net/browse/EMQX-12657

Release version: v/e5.?

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [] Added property-based tests for code which performs user input validation
- [] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [] Schema changes are backward compatible

### CODE REVIEW

File: apps/emqx_bridge_mqtt/test/emqx_bridge_mqtt_SUITE.erl [hunk 1]
```diff
@@ -399,6 +401,56 @@ t_mqtt_conn_bridge_ingress_shared_subscription(_) ->
...
+    snabbkaffe:block_until(
+        fun
+            (#{msg := emqx_bridge_mqtt_connector_tcp_closed}) ->
+                true;
+            (_) ->
+                false
+        end,
+        5000
+    ),
+    Trace = snabbkaffe:collect_trace(),
```
Comment (thalesmg) [inline apps/emqx_bridge_mqtt/test/emqx_bridge_mqtt_SUITE.erl:440]: Nit: wrap the test body in `?check_trace/2` and check the trace there.

File: apps/emqx_bridge_mqtt/test/emqx_bridge_mqtt_SUITE.erl [hunk 2]
```diff
@@ -399,6 +401,56 @@ t_mqtt_conn_bridge_ingress_shared_subscription(_) ->
...
+                <<"local">> => #{
+                    <<"topic">> => <<?INGRESS_LOCAL_TOPIC, "/${topic}">>,
+                    <<"qos">> => <<"${qos}">>,
+                    <<"payload">> => <<"${clientid}">>,
+                    <<"retain">> => <<"${retain}">>
+                }
+            }
+        }
+    ),
+    snabbkaffe:block_until(
```
Comment (thalesmg) [inline apps/emqx_bridge_mqtt/test/emqx_bridge_mqtt_SUITE.erl:431]: Nit: better check that indeed the event happened with `{ok, _} = `, or even better is to set a timetrap in `?check_trace` and don't set a finite timeout here.

i.e.:

```erlang
?check_trace(begin
...
{ok, _} = ?block_until(...), %% `?block_until/1`: no timeout
...
, fun(Trace) -> ... end),
```

File: apps/emqx_bridge_mqtt/src/emqx_bridge_mqtt_connector.erl [hunk 3]
```diff
@@ -508,6 +508,7 @@ connect(Pid, Name) ->
         {ok, _Props} ->
             {ok, Pid};
         {error, Reason} = Error ->
+            log_connect_error_reason(Reason, Name),
             ?SLOG(warning, #{
```
Comment (thalesmg) [inline apps/emqx_bridge_mqtt/src/emqx_bridge_mqtt_connector.erl:512]: This one is now superseded by the line above, it seems.  :scissors:

File: apps/emqx_bridge_mqtt/src/emqx_bridge_mqtt_connector.erl [hunk 4]
```diff
@@ -508,6 +508,7 @@ connect(Pid, Name) ->
         {ok, _Props} ->
             {ok, Pid};
         {error, Reason} = Error ->
+            log_connect_error_reason(Reason, Name),
```
Comment (keynslug) [inline apps/emqx_bridge_mqtt/src/emqx_bridge_mqtt_connector.erl:511]: Q: Shouldn't we aim to beautify all the communication that attempts to describes what just happened, not just log messages? E.g. resource start errors and health check statuses that end up in the dashboard?

**Review by id — APPROVED** [review]

**Comment by emqxqa** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-3470)

**Pull request merged**